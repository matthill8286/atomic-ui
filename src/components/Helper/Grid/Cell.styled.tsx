import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'
import { gutterWidth } from './constants'
import { StyledCellProps, TwelveColumn } from './Grid.interface'

const getXsCols = ({ columns, colsXs }: Pick<StyledCellProps, 'columns' | 'colsXs'>) => {
  const cols = colsXs !== undefined ? colsXs : columns
  if (cols !== undefined) {
    switch (cols) {
      case 0:
        return `
          display: none;
        `
      case 1:
        return css`
          width: calc(25% - ${gutterWidth.xs}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 1;
          }
        `
      case 2:
        return css`
          width: calc(50% - ${gutterWidth.xs}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 2;
          }
        `
      case 3:
        return css`
          width: calc(75% - ${gutterWidth.xs}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 3;
          }
        `
      default:
        return css`
          width: calc(100% - ${gutterWidth.xs}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 4;
          }
        `
    }
  }
}

const getSmMdCols = (md: boolean) => ({
  columns,
  colsSm,
  colsMd,
}: Pick<StyledCellProps, 'columns' | 'colsSm' | 'colsMd'>) => {
  let cols: TwelveColumn | undefined
  if (md) {
    cols = colsMd !== undefined ? colsMd : columns
  } else {
    cols = colsSm !== undefined ? colsSm : columns
  }
  const gutter = md ? gutterWidth.md : gutterWidth.sm
  if (cols !== undefined) {
    switch (cols) {
      case 0:
        return `
          display: none;
        `
      case 1:
        return css`
          width: calc(12.5% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 1;
          }
        `
      case 2:
        return css`
          width: calc(25% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 2;
          }
        `
      case 3:
        return css`
          width: calc(37.5% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 3;
          }
        `
      case 4:
        return css`
          width: calc(50% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 4;
          }
        `
      case 5:
        return css`
          width: calc(62.5% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 5;
          }
        `
      case 6:
        return css`
          width: calc(75% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 6;
          }
        `
      case 7:
        return css`
          width: calc(87.5% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 7;
          }
        `
      default:
        return css`
          width: calc(100% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 8;
          }
        `
    }
  }
}

const getLgXlCols = (xl: boolean) => ({
  columns,
  colsLg,
  colsXl,
}: Pick<StyledCellProps, 'columns' | 'colsLg' | 'colsXl'>) => {
  let cols: TwelveColumn | undefined
  if (xl) {
    cols = colsXl !== undefined ? colsXl : columns
  } else {
    cols = colsLg !== undefined ? colsLg : columns
  }
  const gutter = xl ? gutterWidth.xl : gutterWidth.lg
  if (cols !== undefined) {
    switch (cols) {
      case 0:
        return `
          display: none;
        `
      case 1:
        return css`
          width: calc(8.33333% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 1;
          }
        `
      case 2:
        return css`
          width: calc(16.66667% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 2;
          }
        `
      case 3:
        return css`
          width: calc(25% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 3;
          }
        `
      case 4:
        return css`
          width: calc(33.33333% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 4;
          }
        `
      case 5:
        return css`
          width: calc(41.66667% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 5;
          }
        `
      case 6:
        return css`
          width: calc(50% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 6;
          }
        `
      case 7:
        return css`
          width: calc(58.33333% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 7;
          }
        `
      case 8:
        return css`
          width: calc(66.66667% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 8;
          }
        `
      case 9:
        return css`
          width: calc(75% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 9;
          }
        `
      case 10:
        return css`
          width: calc(83.33333% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 10;
          }
        `
      case 11:
        return css`
          width: calc(91.66667% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 11;
          }
        `
      default:
        return css`
          width: calc(100% - ${gutter}px);

          @supports (display: grid) {
            width: auto;
            grid-column-end: span 12;
          }
        `
    }
  }
}

export const StyledCell = styled.div<StyledCellProps>`
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(${gutterWidth.xs}px / 2);
  width: calc(100% - ${gutterWidth.xs}px);

  @supports (display: grid) {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }

  ${getXsCols}


  ${media.sm} {
    display: ${({ alignHeight }) => (alignHeight ? `flex` : `block`)};
    margin: calc(${gutterWidth.sm}px / 2);
    width: calc(50% - ${gutterWidth.sm}px);

    @supports (display: grid) {
      margin: 0;
      width: auto;
      grid-column-end: span 4;
    }

    ${getSmMdCols(false)}
  }

  ${media.md} {
    display: ${({ alignHeight }) => (alignHeight ? `flex` : `block`)};
    margin: calc(${gutterWidth.md}px / 2);
    width: calc(50% - ${gutterWidth.md}px);

    @supports (display: grid) {
      margin: 0;
      width: auto;
      grid-column-end: span 4;
    }

    ${getSmMdCols(true)}
  }

  ${media.lg} {
    display: ${({ alignHeight }) => (alignHeight ? `flex` : `block`)};
    margin: calc(${gutterWidth.lg}px / 2);
    width: calc(33.33333% - ${gutterWidth.lg}px);

    @supports (display: grid) {
      margin: 0;
      width: auto;
      grid-column-end: span 4;
    }

    ${getLgXlCols(false)}
  }

  ${media.xl} {
    display: ${({ alignHeight }) => (alignHeight ? `flex` : `block`)};
    margin: calc(${gutterWidth.xl}px / 2);
    width: calc(33.33333% - ${gutterWidth.xl}px);

    @supports (display: grid) {
      margin: 0;
      width: auto;
      grid-column-end: span 4;
    }

    ${getLgXlCols(true)}
  }
  
  ${media.ie11} {
    ${({ noMargin }) =>
      noMargin &&
      `
        margin: 0;
    `}
  }
 
  
  ${({ order }) =>
    order &&
    css`
      order: ${order};
    `}

  ${({ align }) => {
    if (align) {
      switch (align) {
        case 'bottom':
          return `
            align-self: flex-end;

            @supports (display: grid) {
              align-self: end;
            }
            `
        case 'middle':
          return `
            align-self: center;
          `
        case 'top':
          return `
          align-self: flex-start;

            @supports (display: grid) {
              align-self: start;
            }
          `
        default:
          return ''
      }
    }
  }}

  
  ${({ justify }) => {
    if (justify) {
      switch (justify) {
        case 'flex-end':
          return `
            justify-self: flex-end;

            @supports (display: grid) {
              justify-self: end;
            }
            `
        case 'stretch':
          return `
            justify-self: stretch;
          `
        case 'center':
          return `
            justify-self: center;
          `
        case 'flex-start':
          return `
          justify-self: flex-start;

            @supports (display: grid) {
              justify-self: start;
            }
          `
        default:
          return ''
      }
    }
  }}

  ${({ alignHeight }) =>
    alignHeight &&
    `display: flex;
      > div {
        flex: 1
      }`}

`
