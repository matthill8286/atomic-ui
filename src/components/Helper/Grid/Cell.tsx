import React, { FunctionComponent } from 'react'
import { StyledCell } from './Cell.styled'
import { CellProps, OffsetProps } from './Grid.interface'

export const Cell: FunctionComponent<CellProps> = ({ tag = 'div', ...rest }) => (
  <StyledCell as={tag} {...rest} />
)

export const Offset: FunctionComponent<OffsetProps> = ({ tag = 'div', ...rest }) => {
  let offsetProps: Omit<OffsetProps, 'tag' | 'children'>

  // According to the unit tests the Offset component defaulted to display: none if no props are given
  if (rest.columns) {
    offsetProps = rest
  } else {
    offsetProps = {
      ...rest,
      colsXs: rest.colsXs || 0,
      colsSm: rest.colsSm || 0,
      colsMd: rest.colsMd || 0,
      colsLg: rest.colsLg || 0,
      colsXl: rest.colsXl || 0,
    }
  }
  return <StyledCell as={tag} {...offsetProps} />
}
