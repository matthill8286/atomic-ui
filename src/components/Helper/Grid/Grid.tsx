import React, { FunctionComponent } from 'react'
import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'
import { fixedColumnValues, gridMargins } from './constants'
import { GridProps, StyledGridProps } from './Grid.interface'

export const handleGridPadding = (noPadding: boolean) => {
  if (noPadding) {
    return css`
      padding: 0;
    `
  } else {
    return css`
      padding: 0 ${gridMargins.xs}px;
      ${media.sm} {
        padding: 0 ${gridMargins.sm}px;
      }
      ${media.md} {
        padding: 0 ${gridMargins.md}px;
      }
      ${media.lg} {
        padding: 0 ${gridMargins.lg}px;
      }
      ${media.xl} {
        padding: 0 ${gridMargins.xl}px;
      }
    `
  }
}

const StyledGrid = styled.div<StyledGridProps>`
  box-sizing: border-box;
  margin: 0 auto;
  ${({ fullWidth }) => !fullWidth && `max-width: 1472px;`};
  ${({ noPadding }) => handleGridPadding(!!noPadding)}

  ${({ align }) => {
    if (align === 'left') {
      return `
    margin-right: auto;
    margin-left: 0;
  `
    }
    if (align === 'right') {
      return `
    margin-left: auto;
    margin-right: 0;
  `
    }
  }}
  ${({ fixedColumnWidth }) =>
    fixedColumnWidth &&
    css`
      width: ${fixedColumnValues.xs.totalWidth}px;
      ${media.sm} {
        width: ${fixedColumnValues.sm.totalWidth}px;
      }
      ${media.md} {
        width: ${fixedColumnValues.md.totalWidth}px;
      }
      ${media.lg} {
        width: ${fixedColumnValues.lg.totalWidth}px;
      }
      ${media.xl} {
        width: ${fixedColumnValues.xl.totalWidth}px;
      }
    `}
`

export const Grid: FunctionComponent<GridProps> = ({ tag = 'div', fullWidth = false, ...rest }) => {
  return <StyledGrid as={tag} fullWidth={fullWidth} {...rest} />
}
