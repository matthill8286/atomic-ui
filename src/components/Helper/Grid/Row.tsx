import React, { FunctionComponent } from 'react'
import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'
import { gutterWidth } from './constants'
import { RowProps, StyledRowProps } from './Grid.interface'

const handleRowMarginBottom = (noMargin: boolean) => {
  if (noMargin) {
    return css`
      margin-bottom: 0;
    `
  } else {
    return css`
      margin-bottom: ${gutterWidth.xs}px;
      ${media.sm} {
        margin-bottom: ${gutterWidth.sm}px;
      }
      ${media.md} {
        margin-bottom: ${gutterWidth.md}px;
      }
      ${media.lg} {
        margin-bottom: ${gutterWidth.lg}px;
      }
      ${media.xl} {
        margin-bottom: ${gutterWidth.xl}px;
      }
    `
  }
}

const StyledRow = styled.div<StyledRowProps>`
  display: flex;
  flex-flow: row wrap;
  align-items: stretch;
  margin: 0;
  ${({ noMargin }) => handleRowMarginBottom(!!noMargin)}

  @supports (display: grid) {
    display: grid;
    grid-gap: ${gutterWidth.xs}px;
    grid-template-columns: repeat(4, minmax(0, 1fr));
  }

  ${media.sm} {
    @supports (display: grid) {
      grid-gap: ${gutterWidth.sm}px;
      grid-template-columns: repeat(8, minmax(0, 1fr));
    }
  }

  ${media.md} {
    @supports (display: grid) {
      grid-gap: ${gutterWidth.md}px;
    }
  }

  ${media.lg} {
    @supports (display: grid) {
      grid-gap: ${gutterWidth.lg}px;
      grid-template-columns: repeat(12, minmax(0, 1fr));
    }
  }

  ${media.xl} {
    @supports (display: grid) {
      grid-gap: ${gutterWidth.xl}px;
    }
  }
`

export const Row: FunctionComponent<RowProps> = ({ tag = 'div', ...rest }) => {
  return <StyledRow as={tag} {...rest} />
}
