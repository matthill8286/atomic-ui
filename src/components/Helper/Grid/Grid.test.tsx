import * as React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { Cell, Offset } from './Cell'
import { Grid } from './Grid'
import { Row } from './Row'

describe('MaterialGrid', () => {
  // list of grid components,
  // useful for iterating over common properties in test for each component
  const components = { Grid, Row, Cell, Offset }

  describe('Grid component', () => {
    it('renders a div per default', () => {
      const tree = renderWithTheme(<Grid />)
      expect(tree).toMatchSnapshot()
    })

    it('renders with Rows and Cells (children)', () => {
      const tree = renderWithTheme(
        <Grid>
          <Row>
            <Cell />
            <Cell />
            <Cell />
            <Cell />
          </Row>
          <Row>
            <Cell />
            <Cell />
            <Cell />
            <Cell />
          </Row>
        </Grid>
      )

      expect(tree).toMatchSnapshot()
    })
  })

  describe('Cell component', () => {
    it('renders with custom responsive column count for view ports', () => {
      const tree = renderWithTheme(
        <Cell colsXs={4} colsSm={8} colsMd={8} colsLg={12} colsXl={12} />
      )

      expect(tree).toMatchSnapshot()
    })

    it('renders with column order', () => {
      const tree = renderWithTheme(<Cell order={2} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders hidden with view port columns count set to 0', () => {
      const tree = renderWithTheme(<Cell colsXs={0} colsSm={0} colsMd={0} colsLg={0} colsXl={0} />)

      expect(tree).toMatchSnapshot()
    })
  })

  describe('Offset component', () => {
    it('renders with custom responsive column count for view ports', () => {
      const tree = renderWithTheme(
        <Cell colsXs={4} colsSm={8} colsMd={8} colsLg={12} colsXl={12} />
      )

      expect(tree).toMatchSnapshot()
    })

    it('renders with column order', () => {
      const tree = renderWithTheme(<Cell order={2} />)
      expect(tree).toMatchSnapshot()
    })

    it('renders hidden with view port columns count set to 0', () => {
      const tree = renderWithTheme(<Cell colsXs={0} colsSm={0} colsMd={0} colsLg={0} colsXl={0} />)

      expect(tree).toMatchSnapshot()
    })

    it('renders hidden with no values, because of default props', () => {
      const tree = renderWithTheme(<Offset />)

      expect(tree).toMatchSnapshot()
    })
  })

  describe('additional customizations', () => {
    describe('with custom tags', () => {
      Object.keys(components).forEach(key => {
        it(`renders ${key} with custom tag name`, () => {
          const Component = components[key]
          const tagName = 'section'

          const tree = renderWithTheme(<Component tag={tagName} />)
          expect(tree).toMatchSnapshot()
        })
      })
    })
  })
})
