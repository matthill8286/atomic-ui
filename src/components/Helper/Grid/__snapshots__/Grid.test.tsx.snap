// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialGrid Cell component renders hidden with view port columns count set to 0 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  display: none;
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    display: none;
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    display: none;
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    display: none;
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    display: none;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<div
  className="c0"
/>
`;

exports[`MaterialGrid Cell component renders with column order 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<div
  className="c0"
  order={2}
/>
`;

exports[`MaterialGrid Cell component renders with custom responsive column count for view ports 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  width: calc(100% - 16px);
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@supports (display:grid) {
  .c0 {
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    width: calc(100% - 16px);
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<div
  className="c0"
/>
`;

exports[`MaterialGrid Grid component renders a div per default 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 1472px;
  padding: 0 16px;
}

@media (min-width:512px) {
  .c0 {
    padding: 0 24px;
  }
}

@media (min-width:752px) {
  .c0 {
    padding: 0 32px;
  }
}

@media (min-width:1008px) {
  .c0 {
    padding: 0 48px;
  }
}

@media (min-width:1232px) {
  .c0 {
    padding: 0 64px;
  }
}

<div
  className="c0"
/>
`;

exports[`MaterialGrid Grid component renders with Rows and Cells (children) 1`] = `
.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
}

.c0 {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 1472px;
  padding: 0 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  margin: 0;
  margin-bottom: 16px;
}

@supports (display:grid) {
  .c2 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c2 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
  }
}

@media (min-width:752px) {
  .c2 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
  }
}

@media (min-width:1008px) {
  .c2 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
  }
}

@media (min-width:1232px) {
  .c2 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

@media (min-width:512px) {
  .c0 {
    padding: 0 24px;
  }
}

@media (min-width:752px) {
  .c0 {
    padding: 0 32px;
  }
}

@media (min-width:1008px) {
  .c0 {
    padding: 0 48px;
  }
}

@media (min-width:1232px) {
  .c0 {
    padding: 0 64px;
  }
}

@media (min-width:512px) {
  .c1 {
    margin-bottom: 16px;
  }
}

@media (min-width:752px) {
  .c1 {
    margin-bottom: 24px;
  }
}

@media (min-width:1008px) {
  .c1 {
    margin-bottom: 24px;
  }
}

@media (min-width:1232px) {
  .c1 {
    margin-bottom: 24px;
  }
}

@supports (display:grid) {
  .c1 {
    display: grid;
    grid-gap: 16px;
    grid-template-columns: repeat(4,minmax(0,1fr));
  }
}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c2"
    />
    <div
      className="c2"
    />
    <div
      className="c2"
    />
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c2"
    />
    <div
      className="c2"
    />
    <div
      className="c2"
    />
  </div>
</div>
`;

exports[`MaterialGrid Offset component renders hidden with no values, because of default props 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  display: none;
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    display: none;
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    display: none;
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    display: none;
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    display: none;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<div
  className="c0"
/>
`;

exports[`MaterialGrid Offset component renders hidden with view port columns count set to 0 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  display: none;
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    display: none;
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    display: none;
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    display: none;
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    display: none;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<div
  className="c0"
/>
`;

exports[`MaterialGrid Offset component renders with column order 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<div
  className="c0"
  order={2}
/>
`;

exports[`MaterialGrid Offset component renders with custom responsive column count for view ports 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  width: calc(100% - 16px);
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@supports (display:grid) {
  .c0 {
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    width: calc(100% - 16px);
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    width: calc(100% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<div
  className="c0"
/>
`;

exports[`MaterialGrid additional customizations with custom tags renders Cell with custom tag name 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<section
  className="c0"
/>
`;

exports[`MaterialGrid additional customizations with custom tags renders Grid with custom tag name 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0 auto;
  max-width: 1472px;
  padding: 0 16px;
}

@media (min-width:512px) {
  .c0 {
    padding: 0 24px;
  }
}

@media (min-width:752px) {
  .c0 {
    padding: 0 32px;
  }
}

@media (min-width:1008px) {
  .c0 {
    padding: 0 48px;
  }
}

@media (min-width:1232px) {
  .c0 {
    padding: 0 64px;
  }
}

<section
  className="c0"
/>
`;

exports[`MaterialGrid additional customizations with custom tags renders Offset with custom tag name 1`] = `
.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  box-sizing: border-box;
  margin: calc(16px / 2);
  width: calc(100% - 16px);
  display: none;
}

@supports (display:grid) {
  .c0 {
    margin: 0;
    width: auto;
    grid-column-end: span 4;
  }
}

@media (min-width:512px) {
  .c0 {
    display: block;
    margin: calc(16px / 2);
    width: calc(50% - 16px);
    display: none;
  }
}

@media (min-width:752px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(50% - 24px);
    display: none;
  }
}

@media (min-width:1008px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    display: none;
  }
}

@media (min-width:1232px) {
  .c0 {
    display: block;
    margin: calc(24px / 2);
    width: calc(33.33333% - 24px);
    display: none;
  }
}

@media screen and (-ms-high-contrast:active),screen and (-ms-high-contrast:none) {

}

<section
  className="c0"
/>
`;

exports[`MaterialGrid additional customizations with custom tags renders Row with custom tag name 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  margin: 0;
  margin-bottom: 16px;
}

@media (min-width:512px) {
  .c0 {
    margin-bottom: 16px;
  }
}

@media (min-width:752px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media (min-width:1008px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media (min-width:1232px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-gap: 16px;
    grid-template-columns: repeat(4,minmax(0,1fr));
  }
}

@media (min-width:512px) {

}

@media (min-width:752px) {

}

@media (min-width:1008px) {

}

@media (min-width:1232px) {

}

<section
  className="c0"
/>
`;
